hudi-client/hudi-client-common/src/main/java/org/apache/hudi/index/bloom/HoodieBloomIndex.java

```diff
@@ -231,12 +231,13 @@ protected List<Pair<String, BloomIndexFileInfo>> loadColumnRangesFromMetaIndex(
     List<Pair<String, BloomIndexFileInfo>> result = new ArrayList<>(fileToColumnStatsMap.size());
 
     for (Map.Entry<Pair<String, String>, HoodieMetadataColumnStats> entry : fileToColumnStatsMap.entrySet()) {
+      ValueMetadata valueMetadata = ValueMetadata.getValueMetadata(entry.getValue().getValueType());
       result.add(Pair.of(entry.getKey().getLeft(),
           new BloomIndexFileInfo(
               partitionAndFileNameToFileId.get(entry.getKey()),
               // NOTE: Here we assume that the type of the primary key field is string
-              unwrapAvroValueWrapper(entry.getValue().getMinValue()).toString(),
-              unwrapAvroValueWrapper(entry.getValue().getMaxValue()).toString()
+              valueMetadata.unwrapValue(entry.getValue().getMinValue()).toString(),
+              valueMetadata.unwrapValue(entry.getValue().getMaxValue()).toString()
```

Comment:
For ensuring backwards compatibility through tests, could you generate test artifacts/tables by using 0.15.0 and 1.0.2 release (and from this PR) with column stats enabled and columns of different primitive and logical types (use MOR with log files), and used that for validating col stats reading and data skipping?  That'll give us much more confidence on compatibility.

Reply:
@jonvex also, when doing the backward compatibility testing, let's enforce stricter validation by comparing the metadata records in column stats index in the Metadata Table written by 0.14 and 1.1 (this PR) using the same set of inserts/updates to make sure the storage bytes are the same.